docs/t8ntool-ref.rst

```diff
@@ -0,0 +1,298 @@
+.. _t8ntool_ref:
+
+###########################################
+Transition Tool
+###########################################
+
+
+
+Command Line Parameters
+=======================
+The command line parameters are used to specify the parameters, input files, and
+output files.
+
+Test Parameters
+---------------
+- **-\\-state.fork** *fork name*
+- **-\\-state.reward** *block mining reward* (appears only in Block tests)
+- **-\\-trace** produce a trace
+
+Input Files
+-----------
+- **-\\-input.alloc** *full path to pretest allocation file*
+- **-\\-input.txs** *full path to transaction file*
+- **-\\-input.env** *full path to environment file*
+
+
+Output Files
+------------
+- **-\\-output.basedir** *directory to write the output files*
+- **-\\-output.result** *file name for test output*
+- **-\\-output.alloc**  *file name for post test allocation file*
```

Comment:
These two can also use `stdout` if user wants them to pass to standard output

Reply:
I also recently added a `output.body`, which outputs a list of rlp transactions

Reply:
Bug report. I use this `evm` version:

```
evm version 1.9.26-unstable-6ea49a04-20210124
```

With an absolute path (`--output.body /tmp/x.rlp`) it produces a binary file.

However, with a relative file path (`--output.basedir <whatever> --output.body <filename>`), no output body file is created.

Reply:
Also, I tried to decode an output.body file using https://codechain-io.github.io/rlp-debugger/, and it didn't work. I'd like to show a sample output in the documentation.

Reply:
Right, I'll look into it. I suppose it might be easier for various purposes if I output it as hex-encoded data instead of binary?

Reply:
I prefer hex-encoded data, but either is fine with me.

Reply:
Actually, I prefer hex-encoded too, plus it's more in line with how other data types are output

Reply:
This one has three transactions: 

```
[user@work evm]$ dir=./testdata/8
[user@work evm]$ ./evm t8n --state.fork=Berlin --input.alloc=$dir/alloc.json --input.txs=$dir/txs.json --input.env=$dir/env.json --trace --output.body=txs.rlp
INFO [02-01|09:06:23.116] Wrote body file                          file=txs.rlp txs=3
INFO [02-01|09:06:23.116] Wrote file                               file=alloc.json
INFO [02-01|09:06:23.117] Wrote file                               file=result.json
```
This is the hex-encoded body: 
```
cat txs.rlp | xxc
f901e8b8c101f8be0180018304ef0094000000000000000000000000000000000000aaaa0180f85bf859940000000000000000000000000000000000000000f842a00000000000000000000000000000000000000000000000000000000000000000a0000000000000000000000000000000000000000000000000000000000000000001a09b83bad6d833b401a9ea725611107e64698e1eac6209f6b7af30b2b62baaece0a0603193024f88264bdb6b864ec88a686823dc1b266598a5740a723acb0e45d19ff86001018304ef0094000000000000000000000000000000000000aaaa028026a05db134df45e75f64425dbae768bebea29f07f72e97694e991908a73bed4a003aa066e7b58c1c7b7b6df1b74e2f318b0fdd7d2f86d0d6e84a8941c9b001f30cbd34b8c101f8be0102018304ef0094000000000000000000000000000000000000aaaa0180f85bf85994000000000000000000000000000000000000aaaaf842a00000000000000000000000000000000000000000000000000000000000000000a0000000000000000000000000000000000000000000000000000000000000000301a0e54909b986c516419130c638c06f744d0635c319bdc09b3ab78c5c2d73c25ec9a0311ab841daa142d7d4c5446fd80f0a5278542c8fa0b47f11b7baa236f205f844
```

Decoding that list of three transactions yields: 
```
 cat txs.rlp | rlpdump
[
  01f8be0180018304ef0094000000000000000000000000000000000000aaaa0180f85bf859940000000000000000000000000000000000000000f842a00000000000000000000000000000000000000000000000000000000000000000a0000000000000000000000000000000000000000000000000000000000000000001a09b83bad6d833b401a9ea725611107e64698e1eac6209f6b7af30b2b62baaece0a0603193024f88264bdb6b864ec88a686823dc1b266598a5740a723acb0e45d19f,
  [
    01,
    01,
    04ef00,
    000000000000000000000000000000000000aaaa,
    02,
    "",
    "&",
    5db134df45e75f64425dbae768bebea29f07f72e97694e991908a73bed4a003a,
    66e7b58c1c7b7b6df1b74e2f318b0fdd7d2f86d0d6e84a8941c9b001f30cbd34,
  ],
  01f8be0102018304ef0094000000000000000000000000000000000000aaaa0180f85bf85994000000000000000000000000000000000000aaaaf842a00000000000000000000000000000000000000000000000000000000000000000a0000000000000000000000000000000000000000000000000000000000000000301a0e54909b986c516419130c638c06f744d0635c319bdc09b3ab78c5c2d73c25ec9a0311ab841daa142d7d4c5446fd80f0a5278542c8fa0b47f11b7baa236f205f844,
]
```

So: `[ blob, rlp-decodable legacy tx, blob]`

When I change the format to use hex-encoded, the example will change slightly, and custom tools like that `xxc` above won't be needed

Reply:
Ok, this has now been fixed. Please try `dir=./testdata/8 && ./evm t8n --state.fork=Berlin --input.alloc=$dir/alloc.json --input.txs=$dir/txs.json --input.env=$dir/env.json --output.basedir=/tmp --output.body=stdout --output.result=stdout`.